client/components/chart/index.js

```diff
@@ -0,0 +1,213 @@
+		this.state = {
+			products,
+			orderedKeys: this.getOrderedKeys( someOrders ).map( d => ( {
+				...d,
+				visible: true,
+				focus: true,
+			} ) ),
+			someOrders,
+			bodyWidth: document.getElementById( 'wpbody' ).getBoundingClientRect().width,
+		};
+	}
+
+	getSomeOrders( products ) {
+		return dummyOrders.map( d => {
+			return Object.keys( d )
+				.filter( key =>
+					products
+						.filter( k => k.selected )
+						.map( k => k.key )
+						.includes( key )
+				)
+				.reduce( ( accum, current ) => ( { ...accum, [ current ]: d[ current ] } ), {} );
+		} );
+	}
+
+	componentDidMount() {
+		window.addEventListener( 'resize', this.updateDimensions );
+	}
+
+	componentWillUnmount() {
+		window.removeEventListener( 'resize', this.updateDimensions );
+	}
+
+	getOrderedKeys( data ) {
+		return [
+			...new Set(
+				data.reduce( ( accum, curr ) => {
+					Object.keys( curr ).forEach( key => key !== 'date' && accum.push( key ) );
+					return accum;
+				}, [] )
+			),
+		]
+			.map( key => ( {
+				key,
+				total: data.reduce( ( a, c ) => a + c[ key ], 0 ),
+			} ) )
+			.sort( ( a, b ) => b.total - a.total );
+	}
+
+	handleChange( event ) {
+		const products = this.state.products.map( d => ( {
+			...d,
+			selected: d.key === event.target.id ? ! d.selected : d.selected,
+		} ) );
+		const someOrders = this.getSomeOrders( products );
+		const orderedKeys = this.getOrderedKeys( someOrders ).map( d => ( {
+			...d,
+			visible: true,
+			focus: true,
+		} ) );
+		this.setState( { products, orderedKeys, someOrders } );
+	}
+
+	handleLegendToggle( event ) {
+		this.setState( {
+			orderedKeys: this.state.orderedKeys.map( d => ( {
+				...d,
+				visible: d.key === event.target.id ? ! d.visible : d.visible,
+			} ) ),
+		} );
+	}
+
+	handleLegendHover( event ) {
+		this.setState( {
+			orderedKeys: this.state.orderedKeys.map( d => {
+				const enterFocus = d.key === event.target.id ? true : false;
+				return {
+					...d,
+					focus: event.type === 'mouseleave' || event.type === 'blur' ? true : enterFocus,
+				};
+			} ),
+		} );
+	}
+
+	updateDimensions() {
+		this.setState( {
+			bodyWidth: document.getElementById( 'wpbody' ).getBoundingClientRect().width,
+		} );
+	}
+
+	render() {
+		const legendDirection =
+			this.state.orderedKeys.length <= 2 && this.state.bodyWidth > 1130 ? 'row' : 'column';
+		const chartDirection =
+			this.state.orderedKeys.length > 2 && this.state.bodyWidth > 1130 ? 'row' : 'column';
+		const legend = (
+			<Legend
+				className={ 'woocommerce_dashboard__widget-legend' }
+				data={ this.state.orderedKeys }
+				handleLegendHover={ this.handleLegendHover }
+				handleLegendToggle={ this.handleLegendToggle }
+				legendDirection={ legendDirection }
+			/>
+		);
+		const margin = {
+			bottom: 50,
+			left: 50,
+			right: 20,
+			top: 0,
+		};
+		return (
+			<Fragment>
+				<Card>
+					<ul>
+						{ this.state.products.map( d => (
+							<li style={ { display: 'inline', 'margin-right': '12px' } }>
```

Comment:
Guessing this might be one of the lines causing the key warnings @psealock mentioned.